### PR TITLE
fix(memory): repair stale profile summary

### LINKED ISSUE #1612: bug(memory): profile summary is pinned to the first explicit fact
## Problem

Memory can successfully ingest and extract new user information while the profile's top-level `summary` remains pinned to the first explicit fact ever extracted. The profile timestamp, `explicit_info`, and `implicit_traits` all advance, but the headline shown under **Summary** can stay as an old one-off instruction indefinitely.

This makes a healthy, updated profile look stale and gives the least representative early fact the most prominent position.

## Observed evidence

Environment:

- Avibe `3.0.13rc1`
- EverOS `1.2.3` (reported by live Memory health)
- `everalgo-user-memory==0.4.0` through the pinned EverOS runtime lock

In a live profile after several accepted `remember` writes:

- Memory health was `ok`
- cascade `pending`, `failed_permanent`, and `failed_retryable` were all `0`
- `profile_timestamp_ms` advanced after later writes
- new explicit information and implicit traits appeared in the profile
- `summary` remained equal to the description of the first low-value explicit item

The profile UI is not discarding updates: `MemoryProfilePanel` renders `summary`, every `explicit_info` item, and every `implicit_traits` item from the same structured response.

## Minimal reproduction

Using the actual `everalgo-user-memory==0.4.0` implementation:

```python
from everalgo.types import Profile
from everalgo.user_memory.profile import _apply_ops

old = Profile.model_validate({
    "owner_id": "u",
    "summary": "Continue",
    "timestamp": 1,
    "explicit_info": [{"description": "Continue"}],
    "implicit_traits": [],
})

updated = _apply_ops(old, [{
    "action": "add",
    "type": "explicit_info",
    "data": {"description": "Prefers evidence-backed engineering decisions"},
}], timestamp=2)

assert len(updated.explicit_info) == 2
assert updated.timestamp == 2
assert updated.summary == "Continue"  # stale headline
```

The isolated run produces:

```text
{'summary': 'Continue',
 'explicit_info': [
   {'description': 'Continue'},
   {'description': 'Prefers evidence-backed engineering decisions'}
 ],
 'timestamp': 2}
```

## Root cause

This is deterministic in `everalgo.user_memory.profile`:

1. UPDATE `add` operations preserve the existing list and append new items.
2. `_apply_ops` rebuilds `summary` after every update.
3. `_build_summary` returns the first non-empty `explicit_info[*].description`; only when no explicit description exists does it inspect implicit traits.

Therefore the first explicit item permanently owns the summary unless it is deleted, updated, or reordered. This is not semantic summarization despite the public field and UI label being `summary`.

The same implementation is still present on current EverAlgo `main` at commit [`20c8622`](https://github.com/EverMind-AI/EverAlgo/blob/20c8622926917d8f0577130f5dcf6fb133c3355d/packages/everalgo-user-memory/src/everalgo/user_memory/profile.py#L649-L663), so upgrading from the currently pinned `0.4.0` alone is not yet a workaround.

## Expected behavior

`summary` should represent the current profile as a whole and should change when later material information changes that profile. A transient first message must not remain the permanent headline after durable preferences and traits have been extracted.

## Acceptance criteria

- `summary` is derived from the current profile as a whole, rather than list position zero.
- A material UPDATE that adds or changes representative information can update `summary`.
- INIT, UPDATE, and compact paths share the same documented summary semantics.
- Regression coverage includes an unrepresentative first explicit item followed by a meaningful later preference/trait.
- Existing profiles can recover on a normal subsequent extraction or a bounded rebuild; users do not need to clear Memory data.
- The Avibe runtime consumes an upstream release containing the fix, or carries a narrowly scoped compatibility fix until one exists.

## Ownership note

The defective summary construction is in the pinned EverAlgo dependency, while Avibe exposes the field as a user-facing Memory profile. This issue should track the product fix end to end, including any upstream report and the runtime pin/update needed to ship it.

### PR DESCRIPTION
**Pull request by cyhhao**
## What changed

Closes #1612.

The EverOS adapter now repairs the known stale-profile shape where the provider's `summary` equals the first explicit profile item while later profile items exist. It selects the latest distinct valid explicit item as the caller-facing summary. Provider-authored summaries that differ from the first item remain unchanged, and this correction leaves the adapter's legacy canonical `text` projection unchanged.

## Why

EverAlgo appends profile updates but rebuilds `summary` from the first explicit description. A one-off first message therefore remains the permanent headline even while the profile timestamp and structured details advance.

## Evidence

- Unit: `tests/test_memory_everos.py` covers stale-summary repair, preservation of a valid provider summary, and preservation of the legacy text projection.
- Contract: `MemoryProfile` and the public profile envelope are unchanged; this is an adapter-only compatibility correction.
- Scenario: no existing scenario catalog covers profile-summary projection; the focused adapter tests exercise the provider response boundary directly.
- Regression: `python3 -m pytest tests/test_memory_everos.py tests/test_memory_runtime.py tests/test_ui_memory_routes.py` -> 430 passed.
- Lint: `ruff check core/memory/everos.py tests/test_memory_everos.py` -> passed.
- Residual manual check: verify the existing user's live profile page after the patched Avibe runtime is installed and the profile is reloaded.

## Known-by-design ledger

- The fix is intentionally read-only at the Avibe boundary. It does not rewrite or delete provider data and does not require users to clear Memory.
- The compatibility heuristic is narrow: summaries that do not match the first explicit item remain provider-authoritative.
- The upstream EverAlgo implementation still needs a semantic summary fix; this PR keeps Avibe usable while the runtime remains pinned to the affected provider behavior.

### CODE REVIEW

**Comment by cyhhao** [comment]
@codex review

File: core/memory/everos.py [hunk 1]
```diff
@@ -1424,6 +1425,38 @@ def _structured_profile(value: Any) -> MemoryProfile | None:
...
+    if summary is None:
+        return None
+
+    if len(explicit_info) > 1 and summary == explicit_info[0].description:
+        for info in reversed(explicit_info[1:]):
+            if info.description != summary:
+                return info.description
+        return summary
+
+    if not explicit_info and len(implicit_traits) > 1 and summary == implicit_traits[0].description:
```
[the remark this answers is not in the document]
Comment (cyhhao) [reply core/memory/everos.py]: Addressed in a9dbfd669. The implicit-only fallback was removed; the compatibility correction now applies only to the explicitly evidenced upstream shape: summary equals the first explicit description while later explicit items exist.

**Comment by cyhhao** [comment]
@codex review
